src/sympc/protocol/aby3/aby3.py

```diff
@@ -237,24 +239,39 @@ def local_decomposition(
         rank = session.rank
         zero = torch.zeros(x.shares[0].shape).type(tensor_type)
 
-        shares = [[zero.clone(), zero.clone()] for i in range(NR_PARTIES)]
+        # Similar to triples, we have instances for the shares generated.
+        share_lst: List[List[List[ReplicatedSharedTensor]]] = []
+
+        input_rst = []
+        if bitwise:
+            ring_bits = get_nr_bits(session.ring_size)  # for bit-wise decomposition
+            for i in range(ring_bits):
+                input_rst.append(x.bit_extraction(i))
+        else:
+            input_rst.append(x)
+
+        for share in input_rst:
+            shares = [[zero.clone(), zero.clone()] for i in range(NR_PARTIES)]
 
-        shares[rank][0] = x.shares[0].clone().type(tensor_type)
+            shares[rank][0] = share.shares[0].clone().type(tensor_type)
+            shares[(rank + 1) % NR_PARTIES][1] = (
+                share.shares[1].clone().type(tensor_type)
+            )
 
-        shares[(rank + 1) % NR_PARTIES][1] = x.shares[1].clone().type(tensor_type)
+            rst_sh = []
+            for i in range(NR_PARTIES):
+                rst = x.clone()
+                rst.shares = shares[i]
+                rst.ring_size = ring_size
+                rst_sh.append(rst)
 
-        rst_list = []
-        for i in range(NR_PARTIES):
-            rst = x.clone()
-            rst.shares = shares[i]
-            rst.ring_size = ring_size
-            rst_list.append(rst)
+            share_lst.append(rst_sh)
 
-        return rst_list
+        return share_lst
```

Comment:
Q: Can't you directly return the ```shares```? the value from line 254?

Reply:
We should RSTensor, the line `254` shares are torch tensor george ?

Reply:
The line `254` shares is a list of torch tensor, we have to return RSTensor for each decomposed share.